library/default-effects-extra.scd

```diff
@@ -412,5 +412,70 @@ DEFAULT EFFECTS EXTRA
 			hpfCutoffFreq
 		);
 		ReplaceOut.ar(out, signal)
-}, [\ir]).add;
+    }, [\ir]).add;
+
+
+    ~dirt.addModule('compressor', { |dirtEvent|
+        var cRatio = ~cRatio ?? 4;
+    	dirtEvent.sendSynth('compressor' ++ ~dirt.numChannels,
+	    	// OPTIONAL
+		    // passing this array of parameters could be left out,
+		    // but it makes it clear what happens
+	    	[
+	    		cThresh: ~cThresh,
+			    cRatio: cRatio,
+	    		out: ~out
+	    	]
+	    )
+    }, { ~cThresh.notNil }); // play synth only if at least one of the two was given
+
+
+    SynthDef("compressor" ++ ~dirt.numChannels, { |out, cThresh, cRatio=4 |
+
+    	var local, signal, signalAmplitude, in, t, o, r, gainReduction, e, cLookaheadDelay, compressedSound, makeupGain;
+    	var cKneelag;
+
+        var cAttack = 0.01;
```

Comment:
you could expose these parameters as controls, then they could be tweaked.

Reply:
Yes, I am aware of this. I can add this in a separate commit in this pull request. I mentioned this topic in the description of this PR: 

 ```
All these parameters could set from the outside, but are static

https://github.com/musikinformatik/SuperDirt/pull/314/files#diff-09ee35f9f1ac58413eab41c2ea003e41c624d845d7fdf1e1b9625836aabd59faR438-R442
```